src/library_webgl.js

```diff
@@ -3957,15 +3957,15 @@ var LibraryGL = {
       return 0;
     }
 
-    var mem = _malloc(length);
+    var mem = _malloc(length), binding = emscriptenWebGLGetBufferBinding(target);
     if (!mem) return 0;
 
-    GL.mappedBuffers[emscriptenWebGLGetBufferBinding(target)] = {
-      offset: offset,
-      length: length,
-      mem: mem,
-      access: access,
-    };
+    if (!GL.mappedBuffers[binding]) GL.mappedBuffers[binding] = {};
+    binding = GL.mappedBuffers[binding];
```

Comment:
It would be shorter, but I think that would generate garbage (since that would assign over the previous {} part). Closure can minify the key names here though.

There are two reasons why I am not considering this code location for strict tight code size savings:
1. This is in an emulation path. Users who utilize strict web code size are likely to also utilize (only) native web features, so this type of OpenGL ES 3 emulation (the FULL_ES3 feature) would not be in that carefully crafted path, and
2. To optimize these emulation functions for code size, we would want to give them the GL_TRACK_ERRORS disabled treatment, as well as optimize other code locations around this for size as well.

Reply:
Oh yeah, I forget that creates now garbage too.. doh!